deployment/action.yml

```diff
@@ -57,76 +64,73 @@ runs:
-        done
-
-        echo "Changeset for creating releases"
-        sos changeset:commit
+        # apply the given definition so the predeploy command will not fail e.g. for a missing app or service
+        if [ ${#SETOPS_DEFINITION} -gt 0 ]; then
+          sos stage:apply -f "$SETOPS_DEFINITION" --auto-approve
+        fi
 
         if [ ${#PREDEPLOY_COMMAND} -gt 0 ]; then
           echo "Run predeploy command"
           first_app=${apps[0]}
-          release_id_of_first_app=${apps_with_release_id[$first_app]}
           resource_parameters=''
           [ ${#PREDEPLOY_COMMAND_CPU} -gt 0 ] && resource_parameters="$resource_parameters --cpu $PREDEPLOY_COMMAND_CPU"
           [ ${#PREDEPLOY_COMMAND_MEMORY} -gt 0 ] && resource_parameters="$resource_parameters --memory $PREDEPLOY_COMMAND_MEMORY"
 
           detach_timeout=''
           [ ${#PREDEPLOY_DETACH_TIMEOUT} -gt 0 ] && detach_timeout="--detach-timeout $PREDEPLOY_DETACH_TIMEOUT"
-          task_id=$(sos --app "$first_app" task:run $detach_timeout --release "$release_id_of_first_app" --entrypoint sh $resource_parameters -- -c "$PREDEPLOY_COMMAND && echo SETOPS_SUCCESS" | tee /dev/stderr | grep -oP '^ID:   \K[a-z0-9]+$')
+          task_id=$(sos --app "$first_app" task:run $detach_timeout --image-tag "${{ inputs.image-tag }}" --entrypoint sh $resource_parameters -- -c "$PREDEPLOY_COMMAND && echo SETOPS_SUCCESS" | tee /dev/stderr | grep -oP '^ID:   \K[a-z0-9]+$')
 
           # Checking the expected SETOPS_SUCCESS log after task finished to ensure that all logs are collected
           sos --app "$first_app" log -n 50 --task "$task_id" | grep SETOPS_SUCCESS > /dev/null
         else
           echo "No predeploy command has been configured, skipping."
         fi
 
+        # apply the updated image-tags
+        COUNT=1
+        apps_with_image_tags=""
         for app in "${apps[@]}"; do
-          release_id=${apps_with_release_id[$app]}
-          sos --app "$app" release:activate "$release_id"
+          apps_with_image_tags="${apps_with_image_tags}${app}=${{ inputs.image-tag }}"
+          if [[ "${#apps[@]}" != "$COUNT" ]]; then
+            apps_with_image_tags="${apps_with_image_tags},"
+          fi
+          ((COUNT+=1))
         done
+        sos stage:apply -t $apps_with_image_tags --auto-approve
 
-        echo "Changeset for activating releases"
-        sos changeset:commit
-
-        echo "Waiting for successful releases"
+        echo "Waiting for successful deployment"
         max_times=${{ inputs.number-of-retries-to-wait-for-successful-deployment }}
         for app in "${apps[@]}"; do
-          release_id=${apps_with_release_id[$app]}
-
           # Checking for health state of HEALTHY when health check is configured or otherwise accept UNKNOWN
-          app_info=$(sos app:info $app)
+          task_info=$(sos --app $app task)
```

Comment:
@kevinscholz Ich glaube, das ist noch falsch, oder? Das wird ja hier genutzt um zu gucken, ob die App einen Healthcheck konfiguriert hat, um zu wissen, ob wir auf Status HEALTY oder nur UNKNOWN warten sollen jeweils.
Wenn ich das richtig sehe, taucht die Info in `task` aber nicht auf, also warten wir so wahrscheinlich bei allen Apps nur auf `UNKNOWN`.

Reply:
Good catch, thanks 🙏 Fixed the issue and the CI output looks good: https://github.com/setopsco/github-actions/actions/runs/4957511659/jobs/8869296431